File: Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    float avg_letters = 100 * ((float) letters / words);
    float avg_sentences = 100 * ((float) sentences / words);
    float index = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}

int count_letters(string text)
{
    int count = 0;
    int length = strlen(text);

    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            count = count + 1;
        }
    }

    return count;
}

int count_words(string text)
{
    int count = 0;
    int length = strlen(text);

    for (int i = 0; i < length; i++)
    {
        if (isblank(text[i]))
        {
            count = count + 1;
        }
    }

    return count + 1;
}

int count_sentences(string text)
{
    int count = 0;
    int length = strlen(text);

    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count = count + 1;
        }
    }

    return count;
}

File: Week 6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
words = len(text.split())
full_stops = len(text.split(". "))
questions = len(text.split("? "))
if (questions > 1):
    sentences = full_stops + questions - 1
else:
    sentences = full_stops
letters = 0
for i in text:
    if (i.isalpha()):
        letters += 1

avg_letters = 100 * (letters/words)
avg_sentences = 100 * (sentences/words)

grade = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)

if (grade > 16):
    print("Grade 16+")
elif (grade < 1):
    print("Before Grade 1")
else:
    print(f"Grade {grade}")
